docs/specification.md

```diff
@@ -61,358 +61,946 @@ For a broader understanding of A2A's purpose and benefits, see [What is A2A?](./
+
+#### 3.1.9. Get Push Notification Config
+
+Retrieves an existing push notification configuration for a task.
+
+**Inputs:**
+
+- `taskId`: Unique identifier of the task
+- `configId`: Unique identifier of the push notification configuration
+
+**Outputs:**
+
+- [`TaskPushNotificationConfig`](#432-taskpushnotificationconfig): The requested configuration
+
+**Behavior:**
+
+The operation MUST return configuration details including webhook URL and notification settings. The operation MUST fail if the configuration does not exist or the client lacks access.
+
+**Transport Implementations:**
+
+- **JSON-RPC**: [`tasks/pushNotificationConfig/get`](#937-push-notification-configuration-methods)
+- **gRPC**: [`GetTaskPushNotificationConfig`](#grpc-push-notification-operations)
+- **HTTP/REST**: [`GET /v1/tasks/{id}/pushNotificationConfigs/{configId}`](#1123-push-notification-configuration)
+
+#### 3.1.10. List Push Notification Configs
+
+Retrieves all push notification configurations for a task.
+
+**Inputs:**
+
+- `taskId`: Unique identifier of the task
+
+**Outputs:**
+
+- Array of [`TaskPushNotificationConfig`](#432-taskpushnotificationconfig) objects
+
+**Behavior:**
+
+The operation MUST return all active push notification configurations for the specified task and MAY support pagination for tasks with many configurations.
+
+**Transport Implementations:**
+
+- **JSON-RPC**: [`tasks/pushNotificationConfig/list`](#937-push-notification-configuration-methods)
+- **gRPC**: [`ListTaskPushNotificationConfig`](#grpc-push-notification-operations)
+- **HTTP/REST**: [`GET /v1/tasks/{id}/pushNotificationConfigs`](#1123-push-notification-configuration)
+
+#### 3.1.11. Delete Push Notification Config
+
+Removes a push notification configuration for a task.
+
+**Inputs:**
+
+- `taskId`: Unique identifier of the task
+- `configId`: Unique identifier of the push notification configuration to delete
+
+**Outputs:**
+
+- Confirmation of deletion (implementation-specific)
+
+**Behavior:**
+
+The operation MUST permanently remove the specified push notification configuration. No further notifications will be sent to the configured webhook after deletion. This operation MUST be idempotent - multiple deletions of the same config have the same effect.
+
+**Transport Implementations:**
+
+- **JSON-RPC**: [`tasks/pushNotificationConfig/delete`](#937-push-notification-configuration-methods)
+- **gRPC**: [`DeleteTaskPushNotificationConfig`](#grpc-push-notification-operations)
+- **HTTP/REST**: [`DELETE /v1/tasks/{id}/pushNotificationConfigs/{configId}`](#1123-push-notification-configuration)
+
+### 3.2. Operation Semantics
+
+#### 3.2.1. Idempotency
+
+- **Get operations** (Get Task, List Tasks, Get Agent Card) are naturally idempotent
+- **Send Message** operations are **NOT** idempotent - each call creates a new task
+- **Cancel Task** operations are idempotent - multiple cancellation requests have the same effect
+
+#### 3.2.2. Error Handling
+
```

Comment:
Currentlt there are separate tables for gRPC, HTTP, etc. 
Maybe add a single canonical error taxonomy table (Abstract Error → Semantic Meaning), then per transport mapping tables referencing that taxonomy. This removes redundancy and eases future changes.

Reply:
I have chosen not to do this for the moment because I am reluctant to open the door to bringing protocol specific elements into the abstractions part of the specification.  I am open to reopening this issue if people feel strongly.